test/vfd.c

```diff
@@ -3915,33 +4032,1949 @@ test_ctl(void)
+ */
+
+static hbool_t
+test_vector_io__write_v_indiv(H5FD_t *lf, uint32_t count, H5FD_mem_t types[], haddr_t addrs[], size_t sizes[],
+                              void *write_bufs[])
+{
+    hbool_t    size_fixed = FALSE;
+    hbool_t    type_fixed = FALSE;
+    hbool_t    result     = TRUE; /* will set to FALSE on failure */
+    hbool_t    verbose    = FALSE;
+    uint32_t   i;
+    size_t     size = ULONG_MAX;
+    H5FD_mem_t type = H5FD_MEM_NTYPES;
+
+    for (i = 0; i < count; i++) {
+
+        SET_SIZE(size_fixed, sizes, size, i);
+
+        SET_TYPE(type_fixed, types, type, i);
+
+        if (H5FDwrite(lf, type, H5P_DEFAULT, addrs[i], size, write_bufs[i]) < 0) {
+
+            if (verbose) {
+
+                HDfprintf(stdout, "%s: HDwrite() failed on entry %d.\n", __func__, i);
+            }
+            result = FALSE;
+            break;
+        }
+    }
+
+    return (result);
+
+} /* end test_vector_io__write_v_indiv() */
+
+/*-------------------------------------------------------------------------
+ *
+ * Function:    test_vector_io__verify_v
+ *
+ * Purpose:     Verify that the read and write buffers of the supplied
+ *              vectors are identical.
+ *
+ * Return:      TRUE if the read and write vectors are identical, and
+ *              FALSE otherwise.
+ *
+ * Programmer:  John Mainzer
+ *              6/21/20
+ *
+ * Changes:     None.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+static hbool_t
+test_vector_io__verify_v(uint32_t count, H5FD_mem_t types[], size_t sizes[], void *write_bufs[],
+                         void *read_bufs[], const char *name)
+{
+    hbool_t     size_fixed = FALSE;
+    hbool_t     type_fixed = FALSE;
+    hbool_t     identical  = TRUE;
+    hbool_t     verbose    = TRUE;
+    uint32_t    i;
+    uint32_t    j;
+    uint32_t    buf_size;
+    char *      w_buf;
+    char *      r_buf;
+    const char *mem_type_names[7] = {"H5FD_MEM_DEFAULT", "H5FD_MEM_SUPER", "H5FD_MEM_BTREE", "H5FD_MEM_DRAW",
+                                     "H5FD_MEM_GHEAP",   "H5FD_MEM_LHEAP", "H5FD_MEM_OHDR"};
+    size_t      size              = ULONG_MAX;
+    H5FD_mem_t  type              = H5FD_MEM_NTYPES;
+
+    i = 0;
+
+    while ((i < count) && (identical)) {
+
+        SET_SIZE(size_fixed, sizes, size, i);
+
+        SET_TYPE(type_fixed, types, type, i);
+
+        buf_size = (uint32_t)(size);
+
+        w_buf = (char *)(write_bufs[i]);
+        r_buf = (char *)(read_bufs[i]);
+
+        j = 0;
+        while ((j < buf_size) && (identical)) {
+
+            if (w_buf[j] != r_buf[j]) {
+
+                identical = FALSE;
+
+                if (verbose) {
+
+                    HDfprintf(stdout, "\n\nread/write buf mismatch in vector/entry");
+                    HDfprintf(stdout, "\"%s\"/%d at offset %d/%d w/r = %c/%c type = %s\n\n", name, i, j,
+                              buf_size, w_buf[j], r_buf[j], mem_type_names[type]);
```

Comment:
You're going to want to go through the code and replace %d with the appropriate PRI macros. Those should be used whenever we specify a type size like uint32_t.

Reply:
Are those supported by all our supported systems?